File: rust/examples/basic.rs
//! Basic example of using the Everruns SDK

use everruns_sdk::{
    AgentCapabilityConfig, CreateAgentRequest, CreateSessionRequest, Error, Everruns,
};
use futures::StreamExt;

#[tokio::main]
async fn main() -> Result<(), Error> {
    // Initialize client from environment
    let client = Everruns::from_env()?;

    // Create an agent with current_time capability
    let agent = client
        .agents()
        .create_with_options(
            CreateAgentRequest::new(
                "example-assistant-rs",
                "You are a helpful assistant for examples.",
            )
            .capabilities(vec![AgentCapabilityConfig::new("current_time")]),
        )
        .await?;
    println!("Created agent:");
    println!("  Name: {}", agent.name);
    println!("  ID: {}", agent.id);
    println!("  Status: {:?}", agent.status);
    println!("  Capabilities: {:?}", agent.capabilities);
    println!("  Created: {}", agent.created_at);

    // Create a session (agent is optional)
    let session = client
        .sessions()
        .create_with_options(
            CreateSessionRequest::new()
                .agent_id(&agent.id)
                .capabilities(vec![AgentCapabilityConfig::new("current_time")]),
        )
        .await?;
    println!("Created session:");
    println!("  ID: {}", session.id);
    println!("  Harness: {}", session.harness_id);
    println!("  Agent: {:?}", session.agent_id);
    println!("  Status: {:?}", session.status);
    println!("  Created: {}", session.created_at);

    // Send a message that uses the current_time capability
    let _message = client
        .messages()
        .create(
            &session.id,
            "What time is it right now? Generate a short joke about the current time.",
        )
        .await?;
    println!("Sent message");

    // Stream events
    let mut stream = client.events().stream(&session.id);
    while let Some(event) = stream.next().await {
        match event {
            Ok(e) => {
                println!("Event: {} - {}", e.event_type, e.id);
                if matches!(
                    e.event_type.as_str(),
                    "output.message.completed" | "turn.completed" | "turn.failed"
                ) {
                    break;
                }
            }
            Err(e) => {
                eprintln!("Error: {}", e);
                break;
            }
        }
    }

    // Clean up
    client.sessions().delete(&session.id).await?;
    client.agents().delete(&agent.id).await?;

    Ok(())
}

File: python/examples/basic.py
"""Basic example of using the Everruns SDK."""

import asyncio

from everruns_sdk import AgentCapabilityConfig, Everruns


async def main():
    # Initialize client from environment
    client = Everruns()

    try:
        # Create an agent with current_time capability
        agent = await client.agents.create(
            name="example-assistant",
            system_prompt="You are a helpful assistant for examples.",
            capabilities=[AgentCapabilityConfig(ref="current_time")],
        )
        print("Created agent:")
        print(f"  Name: {agent.name}")
        print(f"  ID: {agent.id}")
        print(f"  Status: {agent.status}")
        print(f"  Capabilities: {agent.capabilities}")
        print(f"  Created: {agent.created_at}")

        # Create a session (agent is optional)
        session = await client.sessions.create(
            agent_id=agent.id,
            capabilities=[AgentCapabilityConfig(ref="current_time")],
        )
        print("Created session:")
        print(f"  ID: {session.id}")
        print(f"  Harness: {session.harness_id}")
        print(f"  Agent: {session.agent_id}")
        print(f"  Status: {session.status}")
        print(f"  Created: {session.created_at}")

        # Send a message that uses the current_time capability
        message = await client.messages.create(
            session_id=session.id,
            text="What time is it right now? Generate a short joke about the current time.",
        )
        print(f"Sent message: {message.id}")

        # Stream events
        async for event in client.events.stream(session.id):
            print(f"Event: {event.type}")
            if event.type in {"output.message.completed", "turn.completed", "turn.failed"}:
                break

        # Clean up
        await client.sessions.delete(session.id)
        await client.agents.delete(agent.id)

    finally:
        await client.close()


if __name__ == "__main__":
    asyncio.run(main())
